cpp/include/cudf/detail/utilities/algorithm.cuh

```diff
@@ -25,4 +29,47 @@ __device__ __forceinline__ T accumulate(Iterator first, Iterator last, T init, B
   }
   return init;
 }
+
+/**
+ * @brief Utility for calling thrust::copy_if.
+ *
+ * Workaround for thrust::copy_if bug (https://github.com/NVIDIA/thrust/issues/1302)
+ * where it cannot iterate over int-max values `distance(first,last) > int-max`
+ * This calls thrust::copy_if in 2B chunks instead.
+ */
+template <typename InputIterator,
+          typename StencilIterator,
+          typename OutputIterator,
+          typename Predicate>
+OutputIterator copy_if(rmm::exec_policy policy,
+                       InputIterator first,
+                       InputIterator last,
+                       StencilIterator stencil,
+                       OutputIterator result,
+                       Predicate pred)
+{
+  auto const copy_size = std::min(static_cast<std::size_t>(std::distance(first, last)),
+                                  static_cast<std::size_t>(std::numeric_limits<int>::max()));
+
+  auto itr = first;
+  while (itr != last) {
+    auto const copy_end =
+      static_cast<std::size_t>(std::distance(itr, last)) <= copy_size ? last : itr + copy_size;
+    result = thrust::copy_if(policy, itr, copy_end, stencil, result, pred);
+    stencil += std::distance(itr, copy_end);
+    itr = copy_end;
+  }
+  return result;
```

Comment:
Worth moving the implementation into the `cu/cpp` file thus we don't need the extra header inclusions in the current file. Also, this is a host API thus could be placed into `algorithm.hpp` header.

Reply:
This is a template function thus it can't be in `.cu/.cpp` file.